Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/ProfilePictureStepViewController+UIImagePickerControllerDelegate.swift

```
@@ -0,0 +1,39 @@
+//
+// Wire
+// Copyright (C) 2018 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+extension ProfilePictureStepViewController: UIImagePickerControllerDelegate {
+    public func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        guard let image = info[UIImagePickerControllerEditedImage] as? UIImage ?? info[UIImagePickerControllerOriginalImage] as? UIImage else {
+
+            self.dismiss(animated: true)
```

I think we should still try to load the picture from the metadata info, since some pictures are not downloaded to the user device (they exist in iCloud and the device is only holding the preview). We would display the permissions dialogue, but I think this is better than failing silently.

---

FYI: imageFromMediaInfo is our custom method, may be it's worth fixing it and not rewriting?

---

@mikeger I will check the better way of accessing iCloud photos in iOS11.